editor/src/app/inputs/text-input.service.ts

```diff
@@ -48,12 +48,50 @@ export class TextInputService {
   }
 
   updateField(event) {
+
     if (event instanceof KeyboardEvent && (event.key === 'Escape' || event.keyCode === 27)) {
       (event.target as HTMLInputElement).value = this.lastValue;
       (event.target as HTMLInputElement).blur();
       return null;
     }
 
+    if (event.key === 'ArrowDown' || event.keyCode === 40 || event.key === 'ArrowUp' || event.keyCode === 38) {
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
+
+      const unitRegex  = /[a-z%]+/g;
+      const digitRegex = /-?[0-9]\d*(\.\d+)?/g;
+
+      const found = event.target.value.match(unitRegex);
+      const digit = event.target.value.match(digitRegex);
+
+      if (digit !== null && digit.length <= 1 ) {
```

Comment:
You should also check if given unit is in `bigUnits` or `litleUnits`. If there is wrong unit given `10pux` result is `NaNpux` when arrow key is pressed.